### PR TITLE
Move comparison series into math package

### PR DESCRIPTION
**Pull request by chrisprice**
Unfortunately this removes the geomteric zoom behaviour but this is something we're going to revisit with the next release.

### CODE REVIEW

**Comment by tlsim** [comment]
Thanks @chrisprice, I've been meaning to do something similar. 
I've added zooming with rebase of the comparison index to the visual test in tlsim@7cd9a4a786c48d90c01883b24387f7c3f66e910e

File: components/math/percentageChange.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        var percentageChange = function(data) {
+
+            if (data.length === 0) {
+                return [];
+            }
+
+            var initialIndex = percentageChange.initialIndex.value(data);
+            var initialValue = percentageChange.inputValue.value(data[initialIndex]);
+
+            return data.slice(initialIndex, data.length)
```
Comment (ColinEberhardt) [inline components/math/percentageChange.js]: I don't think `percenChange` should perform a slice. Leave that to the user - they might want to produce a chart like this:

http://www.financialsense.com/sites/default/files/users/u244/images/2011/0419-business-cycles.gif

File: visual-tests/test-scripts/series/comparison.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+(function(d3, fc) {
+    'use strict';
+
+    var data = [1, 2, 3].map(function() {
+        return fc.utilities.dataGenerator()
+            .seedDate(new Date(2014, 1, 1))
+            .generate(50);
+    });
+
+    var chartLayout = fc.utilities.chartLayout();
```
Comment (ColinEberhardt) [inline visual-tests/test-scripts/series/comparison.js]: legacy code :-P

File: visual-tests/test-scripts/series/comparison.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        .ticks(5);
+
+    var priceAxis = d3.svg.axis()
+        .scale(priceScale)
+        .orient('left')
+        .ticks(5);
+
+    var percentageChange = fc.math.percentageChange()
+        .inputValue(function(d) { return d.close; })
+        .outputValue(function(d, value) { return (d.percentageChange = value); })
```
Comment (ColinEberhardt) [inline visual-tests/test-scripts/series/comparison.js]: doesn't `(d.percentageChange = value)` return `value`? Does this mean you are adding the percent change value to the series data as a side-effect?

**Comment by ColinEberhardt** [comment]
Looks good - just a few minor comments

**Comment by ColinEberhardt** [comment]
> I've added zooming with rebase of the comparison index to the visual test in tlsim@7cd9a4a

Awesome - thanks @tlsim

File: components/math/percentageChange.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                .map(function(d) {
+                    var result = (percentageChange.inputValue.value(d) - initialValue) / initialValue;
+                    return percentageChange.outputValue.value(d, result);
+                });
+        };
+
+        percentageChange.initialIndex = fc.utilities.functorProperty(0);
```
Comment (tlsim) [inline components/math/percentageChange.js]: `baseIndex` might be a better name.

File: components/math/percentageChange.js (hunk 1 above)
Comment (tlsim) [inline components/math/percentageChange.js]: I'm not sure I understand why slice is needed here.

File: components/math/percentageChange.js (hunk 4 above)
Comment (chrisprice) [inline components/math/percentageChange.js]: Done

File: components/math/percentageChange.js (hunk 1 above)
Comment (chrisprice) [inline components/math/percentageChange.js]: Agreed, done.

File: visual-tests/test-scripts/series/comparison.js (hunk 2 above)
Comment (chrisprice) [inline visual-tests/test-scripts/series/comparison.js]: I was trying to make your regex easier to write :P

File: visual-tests/test-scripts/series/comparison.js (hunk 3 above)
Comment (chrisprice) [inline visual-tests/test-scripts/series/comparison.js]: It's doing both, but it doesn't need to. I think it's a side-effect of an earlier version of the test. I've fixed it and added a test for the other behaviour.

**Comment by chrisprice** [comment]
Thanks @tlsim & @ColinEberhardt, I think I've addressed your comments. Nice work on the visual-test additions as well, I've pulled them into this PR.

**Comment by ColinEberhardt** [comment]
LGTM :+1: thanks @chrisprice

**Pull request merged**